ibind/client/ibkr_client.py

```diff
@@ -62,11 +62,42 @@ def __init__(
         if url is None:
             url = f'https://{host}:{port}{base_route}'
 
+        self._use_oauth = use_oauth
+        url = var.IBIND_OAUTH_REST_URL if self._use_oauth else var.IBIND_REST_URL
+        if url is None:
+            url = f'https://{host}:{port}{base_route}'
         self.account_id = account_id
         super().__init__(url=url, cacert=cacert, timeout=timeout, max_retries=max_retries)
 
+        if self._use_oauth:
+            self.live_session_token, self.live_session_token_expires_ms = req_live_session_token(self)
+
         self.logger.info('#################')
         self.logger.info(f'New IbkrClient(base_url={self.base_url!r}, account_id={self.account_id!r}, ssl={self.cacert!r}, timeout={self._timeout}, max_retries={self._max_retries})')
 
     def make_logger(self):
         self._logger = new_daily_rotating_file_handler('IbkrClient', os.path.join(var.LOGS_DIR, f'ibkr_client_{self.account_id}'))
+
+    def get_headers(
+            self,
+            request_method: str,
+            request_url: str
+            ):
+
+        if (not self._use_oauth) or request_url == self.base_url+'oauth/live_session_token':
+            return {}
+
+        # get headers for endpoints other than live session token request
+        headers = generate_oauth_headers(
+            request_method=request_method,
+            request_url=request_url,
+            live_session_token=self.live_session_token
+        )
+
+        return headers
+
+
+    def test_get_live_session_token(self):
+        live_session_token,live_session_token_expires_ms=req_live_session_token(self)
+        return live_session_token,live_session_token_expires_ms
```

Comment:
Do we need this?

Reply:
Its useful as a test, but I'm not sure where to put it without lots of code.
What do you think? Any ideas where to put it?

Reply:
How do you find this useful? Isn't this a wrapper around:

```python
req_live_session_token(client)
```

We don't do any testing or pre/post processing apart from calling that one function. If one were to use it they may as well just use the code I've listed above, right?

Reply:
We can probably remove this function. It was useful for getting the headers correct and is no longer needed.